core/src/main/java/com/es/core/model/phone/JdbcPhoneDao.java

```diff
@@ -9,9 +10,15 @@
 import java.util.Optional;
 
 @Component
-public class JdbcPhoneDao implements PhoneDao{
+public class JdbcPhoneDao implements PhoneDao {
     @Resource
     private JdbcTemplate jdbcTemplate;
+    private static final String FIND_ALL_SQL =
+            "SELECT * FROM (SELECT * FROM phones OFFSET %d LIMIT %d) as phones_partial " +
+                    "LEFT JOIN phone2color on phones_partial.id = phone2color.phoneId " +
+                    "LEFT JOIN colors on colors.id = phone2color.colorId";
+
+    private final ProductRowCallbackHandler<Phone> handler = new PhoneRowCallbackHandler();
```

Comment:
1. Does that thread-save?. That object has a state, which will be accessed by all threads which use that DAO
2. Check what is happening on the website when you reload productListPage